gateway/health_check.go

```diff
@@ -35,16 +35,33 @@ var (
 	healthCheckLock sync.Mutex
 )
 
-func setCurrentHealthCheckInfo(h map[string]HealthCheckItem) {
+func setCurrentHealthCheckInfo(h *sync.Map) {
 	healthCheckLock.Lock()
 	healthCheckInfo.Store(h)
 	healthCheckLock.Unlock()
 }
 
 func getHealthCheckInfo() map[string]HealthCheckItem {
 	healthCheckLock.Lock()
-	ret := healthCheckInfo.Load().(map[string]HealthCheckItem)
+	syncMap := healthCheckInfo.Load()
 	healthCheckLock.Unlock()
+
+	sm, ok := syncMap.(sync.Map)
+	if !ok {
+		panic("stored health check info is of incorrect type")
```

Comment:
Not sure if a panic is appropriate here. I might be missing some context on how the healthcheck operates though.

Reply:
Thanks for taking a look mate; 1d4890c9 resolves by stepping it down to an error log call with the relevant health check info included